packages/sdk-core/src/modules/breadcrumbs/storage/InMemoryBreadcrumbsStorage.ts

```diff
@@ -20,20 +20,43 @@ export class InMemoryBreadcrumbsStorage implements BreadcrumbsStorage, Backtrace
     private _lastBreadcrumbId: number = TimeHelper.toTimestampInSec(TimeHelper.now());
     private _breadcrumbs: OverwritingArray<Breadcrumb>;
 
-    constructor(maximumBreadcrumbs = 100) {
-        this._breadcrumbs = new OverwritingArray<Breadcrumb>(maximumBreadcrumbs);
+    constructor(private readonly _limits: BreadcrumbsStorageLimits) {
+        this._breadcrumbs = new OverwritingArray<Breadcrumb>(_limits.maximumBreadcrumbs ?? Infinity);
     }
 
     public getAttachments(): BacktraceAttachment<unknown>[] {
         return [this];
     }
 
+    public static factory({ limits }: BreadcrumbsStorageOptions) {
+        return new InMemoryBreadcrumbsStorage(limits);
+    }
+
     /**
      * Returns breadcrumbs in the JSON format
      * @returns Breadcrumbs JSON
      */
     public get(): string {
-        return JSON.stringify([...this._breadcrumbs.values()], jsonEscaper());
+        const sizeLimit = this._limits.maximumBreadcrumbsSize;
+        const breadcrumbs = [...this._breadcrumbs.values()];
+        if (sizeLimit === undefined) {
```

Comment:
why you're applying limits when you want to get a breadcrumb, instead of on add. This won't work correctly for react-native for example - or when a specific framework needs to adjust limits. 

Storing infinite number of breadcrumbs is not a good thing to have

Reply:
I changed the code so it removes the breadcrumbs from the array if the new breadcrumb causes the size to be too large.